grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcDecoratingServiceTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.grpc.GrpcSerializationFormats;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleRequest;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleResponse;
+import com.linecorp.armeria.grpc.testing.Metrics.GaugeRequest;
+import com.linecorp.armeria.grpc.testing.Metrics.GaugeResponse;
+import com.linecorp.armeria.grpc.testing.MetricsServiceGrpc.MetricsServiceBlockingStub;
+import com.linecorp.armeria.grpc.testing.MetricsServiceGrpc.MetricsServiceImplBase;
+import com.linecorp.armeria.grpc.testing.ReconnectServiceGrpc.ReconnectServiceBlockingStub;
+import com.linecorp.armeria.grpc.testing.ReconnectServiceGrpc.ReconnectServiceImplBase;
+import com.linecorp.armeria.grpc.testing.TestServiceGrpc.TestServiceBlockingStub;
+import com.linecorp.armeria.grpc.testing.TestServiceGrpc.TestServiceImplBase;
+import com.linecorp.armeria.protobuf.EmptyProtos.Empty;
+import com.linecorp.armeria.server.DecoratingHttpServiceFunction;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.annotation.Decorator;
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import io.grpc.stub.StreamObserver;
+
+class GrpcDecoratingServiceTest {
```

Comment:
Could you also add simple integration tests for other gRPC stub implementations?
You can see test modules in `it/grpc/*`.

Reply:
Which implementation can I add as other gRPC stub implementations?
Could you give me an example?

Reply:
The following example seems enough for integration tests.
```java
@Decorator(FirstDecorator.class)
private static class TestServiceImpl extends TestServiceImplBase {

    @Override
    @Decorator(MethodFirstDecorator.class)
    public void unaryCall(SimpleRequest request, StreamObserver<SimpleResponse> responseObserver) {
        responseObserver.onNext(SimpleResponse.newBuilder()
                                              .setUsername("test user")
                                              .build());
        responseObserver.onCompleted();
    }
}
```

Reply:
@ikhoon 
Sorry for the late response. 🙇‍♂️
Please review again.

Reply:
Don't worry about that. Take your time.